src/devices/RGBDSensorWrapper/CMakeLists.txt

```diff
@@ -15,17 +15,22 @@ if(NOT SKIP_RGBDSensorWrapper)
   yarp_add_plugin(yarp_RGBDSensorWrapper)
 
   target_sources(yarp_RGBDSensorWrapper PRIVATE RGBDSensorWrapper.cpp
-                                                RGBDSensorWrapper.h
-                                                rosPixelCode.h)
+                                                RGBDSensorWrapper.h)
+  target_sources(yarp_RGBDSensorWrapper PRIVATE $<TARGET_OBJECTS:RGBDRosConversionUtils>)
+
+  target_include_directories(yarp_RGBDSensorWrapper PRIVATE $<TARGET_PROPERTY:RGBDRosConversionUtils,INTERFACE_INCLUDE_DIRECTORIES>)
 
   target_link_libraries(yarp_RGBDSensorWrapper PRIVATE YARP::YARP_os
                                                        YARP::YARP_sig
                                                        YARP::YARP_dev
-                                                       YARP::YARP_rosmsg)
+                                                       YARP::YARP_rosmsg
+                                                       RGBDRosConversionUtils)
```

Comment:
This is what is causing the static builds to fail. You are linking to the `RGBDRosConversionUtils`, this is not required since you are already taking its sources and include directories... Just remove this line...